src/v/cluster/config_frontend.cc

```diff
@@ -87,10 +88,36 @@ ss::future<config_frontend::patch_result> config_frontend::do_patch(
           .errc = errc::waiting_for_recovery, .version = config_version_unset};
     }
 
-    auto data = cluster_config_delta_cmd_data();
+    // do not transform values if the cluster is in the middle of an upgrade.
+    // our aliases might be primary names for the other nodes
```

Comment:
Is this more a defensive forwards looking check? Or is there actually an alias that has the name of a primary name? In general it feels like we should try to avoid colliding primary names and aliases